### PR TITLE
feat(kafka_skin.h): kafka log support

### PR DESCRIPTION
**Pull request by vbirds**
add kafka log support

### CODE REVIEW

File: include/spdlog/sinks/kafka_skin.h [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        if (producer_ != nullptr)
+        {
+            producer_->flush(config_.flush_timeout_ms);
+        }
+    }
+
+protected:
+    void sink_it_(const details::log_msg &msg) override
+    {
+        if (producer_ != nullptr)
```
Comment (gabime) [inline include/spdlog/sinks/kafka_skin.h]: seems producer can never be null according to the ctor, so no need  check if null

**Review by gabime — CHANGES_REQUESTED** [review]

File: include/spdlog/sinks/kafka_skin.h [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        }
+        catch (const std::exception &e)
+        {
+            throw_spdlog_ex(fmt_lib::format("error create kafka instance: {}", e.what()));
+        }
+      }
+
+    ~kafka_skin()
+    {
+        if (producer_ != nullptr)
```
Comment (gabime) [inline include/spdlog/sinks/kafka_skin.h]: seems producer can never be null according to the ctor, so no need check if null

**Review by gabime — CHANGES_REQUESTED** [review]

File: include/spdlog/sinks/kafka_skin.h (hunk 1 above)
Comment (vbirds) [reply include/spdlog/sinks/kafka_skin.h]: Thank you for the corrections，I will fix it

File: include/spdlog/sinks/kafka_skin.h (hunk 2 above)
Comment (vbirds) [reply include/spdlog/sinks/kafka_skin.h]: Thank you for the corrections，I will fix it

**Comment by gabime** [comment]
Thanks @vbirds

**Comment by SpriteOvO** [comment]
@vbirds Is the word `skin` intended? Or just a typo of `sink`?

**Comment by gabime** [comment]
Thanks for noticing. Fixed in 2d57e3b

**Pull request merged**